Code/GraphMol/SmilesParse/CXSmilesOps.cpp

```diff
@@ -414,6 +414,70 @@ std::string quote_string(const std::string &txt) {
   // FIX
   return txt;
 }
+
+std::string get_enhanced_stereo_block(const ROMol &mol,
+                            const std::vector<unsigned int> &atomOrder){
+  std::string res = "";
+  // we need a map from original atom idx to output idx:
+  std::vector<unsigned int> revOrder(mol.getNumAtoms());
+  for(unsigned i=0;i<mol.getNumAtoms();++i){
+    revOrder[atomOrder[i]] = i; 
+  }
+  std::vector<unsigned int> absAts;
+  std::vector<std::vector<unsigned int>> orGps;
+  std::vector<std::vector<unsigned int>> andGps;
+
+  // we want this to be canonical (future proofing)
+  for(const auto &sg : mol.getStereoGroups()) {
+    std::vector<unsigned int> aids;
+    aids.reserve(sg.getAtoms().size());
+    for(const auto at : sg.getAtoms()){
+      aids.push_back(revOrder[at->getIdx()]);
+    }  
+    std::sort(aids.begin(),aids.end());
+    switch(sg.getGroupType()){
+    case StereoGroupType::STEREO_ABSOLUTE:
+      absAts.push_back(aids[0]);
```

Comment:
Maybe this should just be `absAts = aids`?

(This just pushes the first atom in aids, while we want to store the complete vector).

Reply:
Maybe even a move?

Reply:
It's clear that what's there is wrong, but rather than assuming that there will only be on ABS group on the molecule, I'll switch this to be an insert of all the aids in the group.